### PR TITLE
Particle/Vection Fields

### LINKED ISSUE #219: More vection fields
### Feature

Replace hasRain/hasSnow with vectionFields, either an array of strings, or maybe a new module with all the options as curves. The former is probably enough, but the latter would give helpful control of heights.

Options would be rain, snow, embers, sand, leaves, fireflies, and bramble bugs.

### PR DESCRIPTION
**Pull request by MegaPiggy**
## Major features
- New `ParticleFields` module. Add particles to your planet (ex: leaves, rain, snow, fireflies, and more). Resolves #219 

## Minor features
-

## Improvements
-

## Bug fixes
-

### CODE REVIEW

**Comment by MegaPiggy** [comment]
gabagool

File: NewHorizons/Builder/Atmosphere/AirBuilder.cs [hunk 1]
```diff
@@ -57,6 +57,7 @@ public static void Make(GameObject planetGO, Sector sector, PlanetConfig config)
                 airGO.AddComponent<OxygenVolume>()._treeVolume = config.Atmosphere.hasTrees;
             }
 
+            // TODO: figure out what to change "config.Atmosphere.hasRain" to...
```
Comment (JohnCorby) [inline NewHorizons/Builder/Atmosphere/AirBuilder.cs:60]: im assuming this pr is blocked on this?

Comment (MegaPiggy) [reply NewHorizons/Builder/Atmosphere/AirBuilder.cs:60]: yes

Comment (MegaPiggy) [reply NewHorizons/Builder/Atmosphere/AirBuilder.cs:60]: I'm not sure whether I should keep it or just make it obsolete and people will have to figure out how to add the vection field, rain audio, and visor effects themselves.

Comment (xen-42) [reply NewHorizons/Builder/Atmosphere/AirBuilder.cs:60]: Just leave hasRain as is

**Comment by xen-42** [comment]
Just waiting for an example config before reviewing

**Comment by MegaPiggy** [comment]
https://github.com/Outer-Wilds-New-Horizons/nh-examples/pull/28

**Review by xen-42 — APPROVED** [review]

**Pull request merged**